apps/prairielearn/elements/pl-file-preview/pl-file-preview.js

```diff
@@ -59,11 +67,51 @@
             });
         });
 
+        const expandButton = li.querySelector('.file-preview-expand');
+
+        function updateExpandButton(expanded) {
+          toggleExpandPreviewText.textContent = expanded ? 'Collapse' : 'Expand';
+          if (expanded) {
+            expandButton.querySelector('.fa-expand').classList.add('d-none');
+            expandButton.querySelector('.fa-compress').classList.remove('d-none');
+          } else {
+            expandButton.querySelector('.fa-expand').classList.remove('d-none');
+            expandButton.querySelector('.fa-compress').classList.add('d-none');
+          }
+        }
+
+        expandButton.addEventListener('click', (event) => {
+          // Prevent this click from toggling the collapse state.
+          event.stopPropagation();
+
+          // If the preview is not already open, open it in the expanded view.
+          if (!preview.classList.contains('show')) {
+            $(preview).collapse('show');
+            pre.style.maxHeight = 'none';
+            updateExpandButton(true);
+            return;
+          }
+
+          // If the preview is already open, toggle the expanded state.
+          if (pre.style.maxHeight) {
+            pre.style.removeProperty('max-height');
```

Comment:
I see that, if this property is removed, the default CSS is taken from the tag/class instead of the local style. It may be a good idea to add a comment to clarify this.

Reply:
Comment added in b2e18c0, is that sufficient?

Reply:
This looks good.